### ISSUE TITLE
PRSice for training and test

### ISSUE THREAD

**Issue by JuRaGa** [choishingwan/PRSice#101]
Hi,
Thanks for developing PRSice, which is a very useful tool. I was wondering if you could me with a particular issue that I am not sure I can do with PRSice?

In order to make my results more robust, I would like to split my target population into training and test. Therefore, I would estimate the optimal set of SNPs, with the optimal P-value threshold, from the training set. Then, I would derive the PRS from the test set using the estimations from the training model. I don't know if this is possible in PRSice? 

Is there an option to pass the optimal P-value threshold to PRSice, so it just returns the PRS?

Many thanks for your help,
Julia

**Comment by choishingwan** [comment]
Use - -fastscore together with - -no-full and - -bar-levels

(Remove space between the -)
On Fri, 22 Mar 2019 at 10:56 PM, Julia-Ramirez <notifications@github.com<mailto:notifications@github.com>> wrote:

Hi,
Thanks for developing PRSice, which is a very useful tool. I was wondering if you could me with a particular issue that I am not sure I can do with PRSice?

In order to make my results more robust, I would like to split my target population into training and test. Therefore, I would estimate the optimal set of SNPs, with the optimal P-value threshold, from the training set. Then, I would derive the PRS from the test set using the estimations from the training model. I don't know if this is possible in PRSice?

Is there an option to pass the optimal P-value threshold to PRSice, so it just returns the PRS?

Many thanks for your help,
Julia

**Comment by Julia-Ramirez** [comment]
Hi,
Thank you for your quick reply.
Please could you be a bit more specific? Should I use them in the training or in the test calculation? Do those flags require any argument?
Thanks again,

**Comment by choishingwan** [comment]
For the training data set, you'll just use the default parameters. For the test calculation, you can then use 

`--fastscore` which instruct PRSice to only calculate thresholds included in bar-levels
`--no-full` which tell PRSice to ignore p-value of 1 unless it is included in the bar-levels
and `--bar-levels <best threshold>` with <best threshold>=best threshold in the training

**Comment by Julia-Ramirez** [comment]
Hi,
Thank you for your reply. I have run the test calculation with the commands you suggested, but it still included p-value of 1. Then, unfortunately the optimal PRS was the one including all SNVs, corresponding to P-value of 1, so I must have done something wrong.

I am attaching the script and the barplot and summary results, in case you can spot it?
Thank you,
![PRSice_Test_BARPLOT](https://user-images.githubusercontent.com/36241035/54940238-3ea51780-4f22-11e9-98dc-15b2a95f9558.png)


[PRSice_Test.summary.txt](https://github.com/choishingwan/PRSice/files/3004638/PRSice_Test.summary.txt)
[PRSice_test.txt](https://github.com/choishingwan/PRSice/files/3004645/PRSice_test.txt)

**Comment by choishingwan** [comment]
Which version are you using? If it is 2.1.9 or 2.1.10, then I will have to
have a look sometime later this week

**Comment by Julia-Ramirez** [comment]
Hi,
Yes, I am using v2.1.9.
Thanks for your help!

**Comment by choishingwan** [comment]
Could you please try again with v2.1.11 with a clean directory? As I cannot replicate this problem

**Comment by Julia-Ramirez** [comment]
Yes, but I cannot fin the link to download it. Please could you share it with me?
Thanks,

**Comment by choishingwan** [comment]
You can find it here 

https://github.com/choishingwan/PRSice/releases/tag/2.1.11
